--- /srv/home/francesco.conti9/Devel/gvsoc/pulp/pulp/chips/pulp_open_hwpe/cluster.py	2026-06-24 22:33:18.219053754 +0200
+++ /srv/home/francesco.conti9/Devel/gvsoc/core/docs/developer_manual/tutorials/hwpe/create_target/solutions/task4/cluster.py	2026-06-24 22:21:14.656117330 +0200
@@ -19,8 +19,8 @@
 import pulp.cpu.iss.pulp_cores as iss
 from cache.hierarchical_cache import Hierarchical_cache
 
-#### TASK - Change the path from pulp.chips.pulp_open.l1_subsystem to pulp.chips.pulp_open_hwpe.l1_subsystem
-from pulp.chips.pulp_open.l1_subsystem import L1_subsystem
+#### SOLUTION - Change the path from pulp.chips.pulp_open.l1_subsystem to pulp.chips.pulp_open_hwpe.l1_subsystem
+from pulp.chips.pulp_open_hwpe.l1_subsystem import L1_subsystem
 from pulp.event_unit.event_unit_v3 import Event_unit
 from interco.router import Router
 from pulp.mchan.mchan_v7 import Mchan
