--- /srv/home/francesco.conti9/Devel/gvsoc/pulp/pulp/chips/pulp_open_hwpe/pulp_open_board.py	2026-06-24 22:33:18.217053778 +0200
+++ /srv/home/francesco.conti9/Devel/gvsoc/core/docs/developer_manual/tutorials/hwpe/create_target/solutions/task4/pulp_open_board.py	2026-06-24 22:21:56.524595283 +0200
@@ -17,9 +17,9 @@
 import os
 import gvsoc.systree as st
 
-#### TASK - Change the path from pulp.chips.pulp_open.pulp_open to pulp.chips.pulp_open_hwpe.pulp_open
+#### SOLUTION - Change the path from pulp.chips.pulp_open.pulp_open to pulp.chips.pulp_open_hwpe.pulp_open
 
-from pulp.chips.pulp_open.pulp_open import Pulp_open
+from pulp.chips.pulp_open_hwpe.pulp_open import Pulp_open
 from devices.hyperbus.hyperflash import Hyperflash
 from devices.spiflash.spiflash import Spiflash
 from devices.spiflash.atxp032 import Atxp032
