--- /srv/home/francesco.conti9/Devel/gvsoc/pulp/pulp/chips/pulp_open_hwpe/pulp_open.py	2026-06-24 22:33:18.214053814 +0200
+++ /srv/home/francesco.conti9/Devel/gvsoc/core/docs/developer_manual/tutorials/hwpe/create_target/solutions/task4/pulp_open.py	2026-06-24 22:22:33.900125248 +0200
@@ -16,13 +16,13 @@
 
 import gvsoc.systree as st
 
-#### TASK - Change the path from pulp.chips.pulp_open.soc to pulp.chips.pulp_open_hwpe.soc
+#### SOLUTION - Changed from pulp_open to pulp_open_hwpe
 
-from pulp.chips.pulp_open.soc import Soc, SocAttr
+from pulp.chips.pulp_open_hwpe.soc import Soc, SocAttr
 
-#### TASK - Change the path from pulp.chips.pulp_open.cluster to pulp.chips.pulp_open_hwpe.cluster
+#### SOLUTION - Changed from pulp_open to pulp_open_hwpe
 
-from pulp.chips.pulp_open.cluster import Cluster, get_cluster_name
+from pulp.chips.pulp_open_hwpe.cluster import Cluster, get_cluster_name
 from vp.clock_domain import Clock_domain
 from utils.clock_generator import Clock_generator
 from pulp.padframe.padframe_v1 import Padframe
@@ -39,12 +39,12 @@
 
 class Pulp_open(st.Component):
 
-#### TASK - Change the path from soc_config_file='pulp/chips/pulp_open/soc.json to soc_config_file='pulp/chips/pulp_open_hwpe/soc.json'
-#### TASK - Change the path from cluster_config_file='pulp/chips/pulp_open/cluster.json' to cluster_config_file='pulp/chips/pulp_open_hwpe/cluster.json'
-#### TASK - Change the path from padframe_config_file='pulp/chips/pulp_open/padframe.json' to padframe_config_file='pulp/chips/pulp_open_hwpe/padframe.json'
+#### SOLUTION - Change the path from soc_config_file='pulp/chips/pulp_open/soc.json to soc_config_file='pulp/chips/pulp_open_hwpe/soc.json'
+#### SOLUTION - Change the path from cluster_config_file='pulp/chips/pulp_open/cluster.json' to cluster_config_file='pulp/chips/pulp_open_hwpe/cluster.json'
+#### SOLUTION - Change the path from padframe_config_file='pulp/chips/pulp_open/padframe.json' to padframe_config_file='pulp/chips/pulp_open_hwpe/padframe.json'
 
-    def __init__(self, parent, name, attr: PulpOpenAttr, parser, soc_config_file='pulp/chips/pulp_open/soc.json',
-            cluster_config_file='pulp/chips/pulp_open/cluster.json', padframe_config_file='pulp/chips/pulp_open/padframe.json',
+    def __init__(self, parent, name, attr: PulpOpenAttr, parser, soc_config_file='pulp/chips/pulp_open_hwpe/soc.json',
+            cluster_config_file='pulp/chips/pulp_open_hwpe/cluster.json', padframe_config_file='pulp/chips/pulp_open_hwpe/padframe.json',
             use_ddr=False, pim_support=False, pulpnn=False):
         super(Pulp_open, self).__init__(parent, name)
 
