--- /srv/home/francesco.conti9/Devel/gvsoc/pulp/pulp/chips/pulp_open/cluster.py	2026-06-24 19:08:00.900659110 +0200
+++ /srv/home/francesco.conti9/Devel/gvsoc/core/docs/developer_manual/tutorials/hwpe/create_target/task_setup/pulp_open/cluster.py	2026-06-24 21:51:43.283458784 +0200
@@ -1,5 +1,5 @@
 #
-# Copyright (C) 2020 GreenWaves Technologies
+# Copyright (C) 2020-2026 GreenWaves Technologies, ETH Zurich and University of Bologna
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -18,6 +18,8 @@
 import gvsoc.systree as st
 import pulp.cpu.iss.pulp_cores as iss
 from cache.hierarchical_cache import Hierarchical_cache
+
+#### TASK - Change the path from pulp.chips.pulp_open.l1_subsystem to pulp.chips.pulp_open_hwpe.l1_subsystem
 from pulp.chips.pulp_open.l1_subsystem import L1_subsystem
 from pulp.event_unit.event_unit_v3 import Event_unit
 from interco.router import Router
